### PR TITLE
:bug: Decompile files into a java-project for provider to init correctly

### PR DESCRIPTION
**Pull request by pranavgaikwad**
- Place decompiled java files in a dummy java project maintaining the directory structure for packages
- Walk and decompile all JARs/ WARs within
- Update the dependency path to point to the base of exploded app for us to get .jar names only.

### CODE REVIEW

**Review by shawn-hurley — APPROVED** [review]
If we can't bring a file over because it is unparsable, can we log it?

File: provider/internal/java/util.go [hunk 1]
```diff
@@ -13,25 +13,66 @@ import (
...
+	projectPath = filepath.Join(filepath.Dir(archivePath), "java-project")
+
+	err = createJavaProject(ctx, projectPath)
 	if err != nil {
-		return "", err
+		log.Error(err, "failed to create java project", "path", projectPath)
+		return
+	} else {
+		log.V(5).Info("created java project", "path", projectPath)
+	}
```
Comment (djzager) [inline provider/internal/java/util.go:52]: nit. 

```suggestion
	}
	log.V(5).Info("created java project", "path", projectPath)
```

File: provider/internal/java/util.go [hunk 2]
```diff
@@ -13,25 +13,66 @@ import (
...
+// decompile is a function that extracts the contents of a Java archive (.jar|.war|.ear) and
+// decompiles any .class files found using the fernflower decompiler into java project location
+// maintaining the tree. swallows decomp and copy errors, returns others
+func decompile(ctx context.Context, log logr.Logger, archivePath, projectPath string) (string, error) {
 	// Create the destDir directory using the same permissions as the Java archive file
 	// java.jar should become java-jar-decompiled
-	destDir := filepath.Join(path.Dir(location), strings.Replace(path.Base(location), ".", "-", -1) + "-decompiled")
-	err = os.MkdirAll(destDir, fileInfo.Mode())
+	destDir := filepath.Join(path.Dir(archivePath), strings.Replace(path.Base(archivePath), ".", "-", -1)+"-decompiled")
+	err := os.MkdirAll(destDir, 0755)
```
Comment (djzager) [inline provider/internal/java/util.go:70]: This should make it so you only set the execute bit for everyone on directories in the case it's missing:

```suggestion
	err := os.MkdirAll(destDir, fileInfo.Mode()|0111)
```

File: provider/internal/java/util.go [hunk 3]
```diff
@@ -72,17 +119,44 @@ func decompileJava(ctx context.Context, log logr.Logger, location string) (strin
...
+				ctx, "java", "-jar", "/bin/fernflower.jar", filePath, path.Dir(destPath))
 			err := cmd.Run()
 			if err != nil {
-				log.Error(err, "Failed to decompile file", filePath)
+				log.Error(err, "failed to decompile file", "file", filePath)
 			} else {
-				log.Info("Decompiled file", filePath)
+				log.V(8).Info("decompiled file", "file", filePath)
+			}
+		} else if strings.HasSuffix(f.Name, ".jar") || strings.HasSuffix(f.Name, ".war") {
```
Comment (djzager) [inline provider/internal/java/util.go:142]: Should use the constants.

File: provider/internal/java/util.go [hunk 4]
```diff
@@ -72,17 +119,44 @@ func decompileJava(ctx context.Context, log logr.Logger, location string) (strin
 			return "", err
 		}
 
-		// If we found a class file, decompile it
+		// If we found a class file, decompile it to java project path
 		if strings.HasSuffix(f.Name, ".class") {
```
Comment (djzager) [inline provider/internal/java/util.go:123]: I should have made this a constant when I originally wrote it.

**Review by djzager — COMMENTED** [review]
Some nitpicks but it all looks gravy to me.

**Comment by pranavgaikwad** [comment]
also made it a tad bit faster by not decompiling same file again

**Review by djzager — APPROVED** [review]

**Pull request merged**